afew/MailMover.py

```diff
@@ -59,7 +62,7 @@ def move(self, maildir, rules):
             messages = notmuch.Query(self.db, main_query).search_messages()
             for message in messages:
```

Comment:
can we add below per-message code into a self-contained staticmethod? This would greatly improve readability and testability.

Reply:
Not really sure to fully understand, but if you talk about export the whole per-message process I agree, I think the whole class need some refactor, for example the move() method is also searching the messages which is problematic to add tests and keep simple.

So I have project to refactor the class to be more simple, and support some unit tests, but as the mail mover is a big feature I think it would be safer to do that after.

Reply:
Ah yes I agree the thing is I feel it would be simpler to use the function `move()` to do the actual moving operation, and create an other function like `getMessages()` to do the notmuch query and pass the message list to the move(), that way it could be easy tested by using mock data.